hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/math/CeilFunctionIntegrationTest.java

```diff
@@ -0,0 +1,190 @@
+
+        checkColumn(Double.POSITIVE_INFINITY, SqlColumnType.DOUBLE, Double.POSITIVE_INFINITY);
+        checkColumn(Double.NEGATIVE_INFINITY, SqlColumnType.DOUBLE, Double.NEGATIVE_INFINITY);
+        checkColumn(Double.NaN, SqlColumnType.DOUBLE, Double.NaN);
+
+        map.clear();
+        map.put(0, new ExpressionValue.IntegerVal());
+        assertNull(execute("field1", SqlColumnType.INTEGER));
+
+        checkColumnFailure("bad", SqlErrorCode.DATA_EXCEPTION, "Cannot convert VARCHAR to DECIMAL");
+        checkColumnFailure('b', SqlErrorCode.DATA_EXCEPTION, "Cannot convert VARCHAR to DECIMAL");
+        checkColumnFailure(new ExpressionValue.ObjectVal(), SqlErrorCode.PARSING, "Cannot apply 'CEIL' to arguments of type 'CEIL(<OBJECT>)'");
+    }
+
+    private void checkColumn(Object value, SqlColumnType expectedType, Object expectedResult) {
+        map.clear();
+        map.put(0, value);
+
+        Object res = execute("this", expectedType);
+        assertEquals(expectedResult, res);
+    }
+
+    private void checkColumnFailure(Object value, int expectedErrorCode, String expectedErrorMessage) {
+        map.clear();
+        map.put(0, value);
+
+        checkFailure("this", expectedErrorCode, expectedErrorMessage);
+    }
+
+    @Test
+    public void testParameter() {
+        map.put(0, 0);
+
+        checkParameter((byte) 1, BigDecimal.ONE);
+        checkParameter((short) 1, BigDecimal.ONE);
+        checkParameter(1, BigDecimal.ONE);
+        checkParameter(1L, BigDecimal.ONE);
+        checkParameter(0.9f, BigDecimal.ONE);
+        checkParameter(0.9d, BigDecimal.ONE);
+        checkParameter(BigInteger.ONE, BigDecimal.ONE);
+        checkParameter(new BigDecimal("0.9"), BigDecimal.ONE);
+
+        checkParameter("0.9", BigDecimal.ONE);
+        checkParameter('1', BigDecimal.ONE);
+
+        assertNull(execute("?", SqlColumnType.DECIMAL, new Object[] { null }));
+
+        checkFailure("?", SqlErrorCode.DATA_EXCEPTION, "Failed to convert parameter at position 0 from VARCHAR to DECIMAL", "bad");
+        checkFailure("?", SqlErrorCode.DATA_EXCEPTION, "Failed to convert parameter at position 0 from VARCHAR to DECIMAL", 'b');
+        checkFailure("?", SqlErrorCode.DATA_EXCEPTION, "Failed to convert parameter at position 0 from OBJECT to DECIMAL", new ExpressionValue.ObjectVal());
+    }
+
+    private void checkParameter(Object param, Object expectedResult) {
+        Object res = execute("?", SqlColumnType.DECIMAL, param);
+        assertEquals(res, expectedResult);
+    }
+
+    @Test
+    public void testLiteral() {
+        map.put(0, 0);
+
+        checkLiteral(1, SqlColumnType.TINYINT, (byte) 1);
+
+        checkLiteral("null", SqlColumnType.DECIMAL, null);
+        checkLiteral("1.1", SqlColumnType.DECIMAL, new BigDecimal("2"));
+        checkLiteral("1.1E0", SqlColumnType.DOUBLE, 2d);
+
+        checkFailure("'bad'", SqlErrorCode.PARSING, "Literal ''bad'' can not be parsed to type 'DECIMAL'");
```

Comment:
We should test a good string literal too.

Reply:
Added missing tests to CEIL/FLOOR test classes